components/translate/core/browser/resources/brave_translate.js

```diff
@@ -0,0 +1,77 @@
+const securityOriginHost = new URL(securityOrigin).host;
+const redirectToSecurityOrigin = (url) => {
+  let new_url = new URL(url);
+  if (useGoogleTranslateEndpoint && new_url.pathname === '/translate_a/t') {
+    new_url.host = 'translate.googleapis.com';
+
+    // Remove API key
+    new_url.searchParams.set('key', '');
+
+    return new_url.toString();
+  }
+  new_url.host = securityOriginHost;
+  return new_url.toString();
+};
+
+const emptySvgDataUrl = 'data:image/svg+xml;base64,' +
+  btoa('<svg xmlns="http://www.w3.org/2000/svg"/>');
+
+function processJavascript(text) {
+  text = text.replaceAll('"//"+po+"/gen204?"+Bo(b)',
+    '"' + emptySvgDataUrl + '"');
+  text = text.replaceAll(
+    "https://www.gstatic.com/images/branding/product/1x/translate_24dp.png",
+    emptySvgDataUrl);
+  return text;
+}
+
+function processCSS(text) {
+  text = text.replaceAll('//www.gstatic.com/images/branding/product/2x/translate_24dp.png',
+    emptySvgDataUrl);
+  return text;
+}
+
+if (typeof XMLHttpRequest.prototype.realOpen === 'undefined') {
+  XMLHttpRequest.prototype.realOpen = XMLHttpRequest.prototype.open;
+  XMLHttpRequest.prototype.open = function (method, url, async = true,
```

Comment:
We override this for all requests when the page translate is actually triggered.

Frankly speaking, the current scripts uses relative paths => It seems we can remove this code now and stash it for the case of script behaviour changing.

Reply:
i think you want to add some filtering so that XHR.open behaves the same as originally for requests which are unrelated to google translate: https://github.com/brave/brave-core/pull/13845/files#r907905698